packages/core/src/routes/hook.ts

```diff
@@ -1,29 +1,76 @@
-import { koaAdapter, RequestError } from '@withtyped/server';
-import type { MiddlewareType } from 'koa';
-import koaBody from 'koa-body';
+import { generateStandardId } from '@logto/core-kit';
+import { Hooks } from '@logto/schemas';
+import { z } from 'zod';
 
-import LogtoRequestError from '#src/errors/RequestError/index.js';
+import koaGuard from '#src/middleware/koa-guard.js';
 
 import type { AuthedRouter, RouterInitArgs } from './types.js';
 
-// Organize this function if we decide to adopt withtyped eventually
-const errorHandler: MiddlewareType = async (_, next) => {
-  try {
-    await next();
-  } catch (error: unknown) {
-    if (error instanceof RequestError) {
-      throw new LogtoRequestError(
-        { code: 'request.general', status: error.status },
-        error.original
-      );
+export default function hookRoutes<T extends AuthedRouter>(
+  ...[router, { queries }]: RouterInitArgs<T>
+) {
+  const { findAllHooks, findHookById, insertHook, updateHookById, deleteHookById } = queries.hooks;
+
+  router.get('/hooks', async (ctx, next) => {
```

Comment:
nit: response guard?

Reply:
yes, i'll do this today. find some issues on schema generator